nemo-physical/src/columnar/operations/columnscan_restrict_values.rs

```diff
@@ -0,0 +1,355 @@
+use super::super::traits::columnscan::{ColumnScan, ColumnScanCell};
+use crate::datatypes::ColumnDataType;
+use std::{fmt::Debug, ops::Range};
+
+/// Concrete value a interval bound can take.
+#[derive(Debug, Clone)]
+pub enum FilterValue<T>
+where
+    T: Clone,
+{
+    /// Interval bound is given as a value pointed by a column scan with the given index.
+    Column(usize),
+    /// Interval bound by the given constant.
+    Constant(T),
+}
+
+impl<T> FilterValue<T>
+where
+    T: Clone,
+{
+    /// Return the column index this value refers to.
+    /// Returns None if this is a constant.
+    pub fn column_index(&self) -> Option<usize> {
+        if let Self::Column(index) = self {
+            Some(*index)
+        } else {
+            None
+        }
```

Comment:
Please use an explicit `match`, so we will not silently do the wrong thing when a new variant is added to the enum.